lib/type/string.ex

```diff
@@ -75,6 +78,43 @@ defmodule Litmus.Type.String do
     {:ok, params}
   end
 
+  @spec regex_validate(t, binary, map) :: {:ok, map} | {:error, binary}
+  defp regex_validate(%__MODULE__{regex: %{pattern: regex, error_message: msg}}, field, params)
+       when regex != nil and msg != nil do
+    if Map.has_key?(params, field) and !Regex.match?(regex, params[field]) do
+      {:error, msg}
+    else
+      {:ok, params}
+    end
+  end
+
+  defp regex_validate(%__MODULE__{regex: %{pattern: regex, error_message: nil}}, field, params)
+       when regex != nil do
+    if Map.has_key?(params, field) and !Regex.match?(regex, params[field]) do
+      {:error, "#{field} must be in a valid format"}
+    else
+      {:ok, params}
+    end
+  end
```

Comment:
I think we could cut this from 3 functions down to 2 like so:

```elixir
defp regex_validate(%__MODULE__{regex: %{pattern: nil}, field, params) when do
  {:ok, params}
end

defp regex_validate(%__MODULE__{regex: regex, field, params) do
  if Map.has_key?(params, field) and !Regex.match?(regex.pattern, params[field]) do
    {:error, Map.get(regex, :error_message, "#{field} must be in a valid format")}
  else
    {:ok, params}
  end
end
```

That way, we cut down on some duplicate code. What do you think?

Also, I think it is often easier to rule out cases like the `pattern: nil` in the first function header. That way, we don't have to check if it's `nil` in a when clause of all other function headers for this function.